radix-engine-tests/tests/wasm_non_mvp.rs

```diff
@@ -0,0 +1,124 @@
+// since rust 1.70.0
+// see: https://github.com/rust-lang/rust/issues/109807
+macro_rules! assert_sign_extensions {
+    ($type:expr, $instruction:expr, $input:expr, $output:expr) => {
+        paste! {
+            #[test]
+            fn [<test_wasm_non_mvp_sign_extensions_ $type _ $instruction>]() {
+                // Arrange
+                let value_kind = BasicValueKind::[<$type:upper>].as_u8().to_string();
+                let slice_len = (
+                    1 +                           // prefix byte
+                    1 +                           // value kind byte
+                    std::mem::size_of::<$type>()  // value bytes
+                ).to_string();
+                let input = $input as $type;
+
+                // Act
+                let code = wat2wasm(&include_str!("wasm/sign_extensions.wat")
+                        .replace("${base}", stringify!($type))
+                        .replace("${instruction}", $instruction)
+                        .replace("${initial}", &input.to_string())
+                        .replace("${value_kind}", &value_kind)
+                        .replace("${slice_len}", &slice_len));
+
+                let mut test_runner = TestRunner::builder().build();
+                let package_address = test_runner.publish_package(
+                    code,
+                    single_function_package_definition("Test", "f"),
+                    BTreeMap::new(),
+                    OwnerRole::None,
+                );
+                let manifest = ManifestBuilder::new()
+                    .lock_fee(test_runner.faucet_component(), 10.into())
+                    .call_function(package_address, "Test", "f", manifest_args!())
+                    .build();
+                let receipt = test_runner.execute_manifest(manifest, vec![]);
+
+                // Assert
+                let outcome: $type = receipt.expect_commit(true).output(1);
+                assert_eq!(outcome, $output as $type);
+            }
+        }
+    };
+}
+
+assert_sign_extensions!(i32, "extend8_s", 0x44332211, 0x11);
+assert_sign_extensions!(i32, "extend16_s", 0x44332211, 0x2211);
+assert_sign_extensions!(i64, "extend8_s", 0x665544332211, 0x11);
+assert_sign_extensions!(i64, "extend16_s", 0x665544332211, 0x2211);
+assert_sign_extensions!(i64, "extend32_s", 0x665544332211, 0x44332211);
+
+#[test]
+fn test_wasm_non_mvp_expect_sign_ext_from_rust_code() {
+    // Arrange
+    let (code, _) = Compile::compile("./tests/blueprints/wasm_non_mvp");
+
+    assert!(WasmModule::init(&code).unwrap().contains_sign_ext_ops())
+}
+
+// Below tests verify WASM "mutable-global" feature, which allows importing/exporting mutable globals.
+// more details:
+// - https://github.com/WebAssembly/mutable-global/blob/master/proposals/mutable-global/Overview.md
+
+// NOTE!
+//  We test only WASM code, because Rust currently does not use the WASM "global" construct for globals
```

Comment:
I spent a significant amount of time trying to test this feature in rust. But unfortunately Rust does not support it ATM. And yes, referenced issues are pretty old, but haven't found anything more up to date.

Taking above into account I am not really sure if testing it for WASM only makes sense. But we can always disable or remove it.